browser/ui/views/brave_actions/brave_shields_action_view.cc

```diff
@@ -0,0 +1,131 @@
+#include "components/prefs/pref_service.h"
+#include "extensions/common/constants.h"
+#include "third_party/skia/include/core/SkBitmap.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/resource/resource_bundle.h"
+#include "ui/gfx/geometry/rect.h"
+#include "ui/gfx/geometry/skia_conversions.h"
+#include "ui/gfx/image/canvas_image_source.h"
+#include "ui/gfx/image/image.h"
+#include "ui/gfx/image/image_skia.h"
+#include "ui/gfx/image/image_skia_rep_default.h"
+#include "ui/views/animation/ink_drop_impl.h"
+#include "ui/views/controls/button/label_button_border.h"
+#include "ui/views/controls/highlight_path_generator.h"
+#include "ui/views/view.h"
+#include "ui/views/view_class_properties.h"
+
+namespace {
+
+class BraveShieldsActionViewHighlightPathGenerator
+    : public views::HighlightPathGenerator {
+ public:
+  BraveShieldsActionViewHighlightPathGenerator() = default;
+
+  // HighlightPathGenerator
+  SkPath GetHighlightPath(const views::View* view) override {
+    return static_cast<const BraveShieldsActionView*>(view)->GetHighlightPath();
+  }
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(BraveShieldsActionViewHighlightPathGenerator);
+};
+
+}  // namespace
+
+BraveShieldsActionView::BraveShieldsActionView(
+    BraveShieldsActionView::Delegate* delegate)
+    : LabelButton(base::BindRepeating(&BraveShieldsActionView::ButtonPressed,
+                                      base::Unretained(this)),
+                  std::u16string()),
+      delegate_(delegate) {
+  auto* ink_drop = views::InkDrop::Get(this);
+  ink_drop->SetMode(views::InkDropHost::InkDropMode::ON);
+  ink_drop->SetBaseColorCallback(base::BindRepeating(
+      [](views::View* host) { return GetToolbarInkDropBaseColor(host); },
+      this));
+
+  SetAccessibleName(l10n_util::GetStringUTF16(IDS_BRAVE_UI_BRAVE_REWARDS));
+  SetHasInkDropActionOnClick(true);
+  SetHorizontalAlignment(gfx::ALIGN_CENTER);
+  ink_drop->SetVisibleOpacity(kToolbarInkDropVisibleOpacity);
+  // Create badge-and-image source like an extension icon would
+  auto preferred_size = GetPreferredSize();
+  ui::ResourceBundle& rb = ui::ResourceBundle::GetSharedInstance();
+  std::unique_ptr<IconWithBadgeImageSource> image_source(
+      new BraveActionIconWithBadgeImageSource(preferred_size));
+  // Set icon on badge using actual extension icon resource
+  gfx::ImageSkia image;
+
+  const SkBitmap bitmap =
+      rb.GetImageNamed(IDR_BRAVE_SHIELDS_ICON_64).AsBitmap();
+  float scale = static_cast<float>(bitmap.width()) / kBraveActionGraphicSize;
+  image.AddRepresentation(gfx::ImageSkiaRep(bitmap, scale));
+  image_source->SetIcon(gfx::Image(image));
+  // TODO(nullhook): Create badge and set text on badge via pref text
+  gfx::ImageSkia icon(
+      gfx::Image(gfx::ImageSkia(std::move(image_source), preferred_size))
+          .AsImageSkia());
+  // Use badge-and-icon source for button's image in all states
+  SetImage(views::Button::STATE_NORMAL, icon);
```

Comment:
Let's use `SetImageModel()` in the new code. see https://source.chromium.org/chromium/chromium/src/+/main:ui/views/controls/button/label_button.h;l=50

Reply:
nice one, done.